pkg/signup/service/signup_service.go

```diff
@@ -318,7 +318,7 @@ func (s *ServiceImpl) PhoneNumberAlreadyInUse(userID, e164PhoneNumber string) er
 		return errors3.NewInternalError(err, "failed listing userSignups")
 	}
 	for _, signup := range userSignupList.Items {
-		if signup.Name != userID {
+		if signup.Spec.UserID != userID && signup.Annotations[v1alpha1.UserSignupUserEmailAnnotationKey] != email {
```

Comment:
So we missed another place with signup.Name? Can we try to add a test for this case? Ideally with an e2e tests too.

Reply:
Sure, we can add a test here.  It should be for a UserSignup with a non DNS-1123 compliant user ID though.